validate.sh

```diff
@@ -0,0 +1,183 @@
+    echo -e "-r, --runs    \t set the total number of operations executed by each thread of each benchmark, default is ${RUNS_PER_THREAD}"
+    echo -e "-w, --max_work\t set the amount of workload (i.e. dummy loop iterations among two consecutive operations of the benchmarked object), default is 64"
+    echo -e ""
+    echo -e "-h, --help    \t displays this help and exits"
+    echo -e ""
+}
+
+COLOR_PASS="[ \e[32mPASS\e[39m ]"
+COLOR_FAIL="[ \e[31mFAIL\e[39m ]"
+
+declare -a uobjects=("ccsynchbench.run"                     "dsmsynchbench.run" "hsynchbench.run" "oscibench.run"      "simbench.run")
+declare -a queues=(  "ccqueuebench.run" "clhqueuebench.run" "dsmqueuebench.run" "hqueuebench.run" "osciqueuebench.run" "simqueuebench.run")
+declare -a stacks=(  "ccstackbench.run" "clhstackbench.run" "dsmstackbench.run" "hstackbench.run" "oscistackbench.run" "simstackbench.run")
+
+declare -a validate_objects=()
+
+validate_objects+=${uobjects[@]}
+validate_objects+=${queues[@]}
+validate_objects+=${stacks[@]}
+
+if [ "$1" = "--help" ] || [ "$1" = "-h" ]; then
+    usage;
+    exit;
+fi
+
+while [ "$1" != "" ]; do
+    PARAM=`echo $1 | awk -F= '{print $1}'`
+    VALUE=`echo $1 | awk -F= '{print $2}'`
+    case $PARAM in
+        -h | --help)
+            usage;
+            exit;
+            ;;
+        -s | --step_threads)
+            STEP_THREADS=$VALUE
+            STEP_SELETCTED=1
+            ;;
+        -t | --max_threads)
+            MAX_PTHREADS=$VALUE
+            ;;
+        -f | --fibers)
+            FIBERS="-f $VALUE"
+            ;;
+        -w | --max_work)
+            WORKLOAD="-w $VALUE"
+            ;;
+        -n | --numa_nodes)
+            NUMA_NODES="-n $VALUE"
+            ;;
+        -r | --runs)
+            RUNS_PER_THREAD=$VALUE
+            ;;
+        -*)
+            echo "ERROR: unknown parameter \"$PARAM\""
+            usage
+            exit 1
+            ;;
+	*)
+	    FILE=$PARAM
+	    ;;
+    esac
+    shift
+done
+
+# Calculate step according to user preferences
+if [ $STEP_SELETCTED -eq 0 ]; then
+    if [ $MAX_PTHREADS -gt 8 ]; then 
+        STEP_THREADS=$(($MAX_PTHREADS / 8))
+    fi
+fi
+
+# Fill the array with the selected number of threads
+PTHREADS_ARRAY=()
+
+# Add set of iterations for 1 thread, if needed
+if [ $STEP_THREADS -ne 1 ]; then
+    PTHREADS_ARRAY+=(1)
+fi
+# Add intermediate sets
+for (( PTHREADS=$STEP_THREADS; PTHREADS<$MAX_PTHREADS; PTHREADS+=$STEP_THREADS ));do
+    PTHREADS_ARRAY+=($PTHREADS)
+done
+# Add set of iterations for max threads
+PTHREADS_ARRAY+=($MAX_PTHREADS)
+
+echo "Compiling the sources..."
+make clean debug > $BUILD_LOG
+
+# Run the selected number of threads
+for PTHREADS in "${PTHREADS_ARRAY[@]}"; do
+    printf "\n\e[36mValidating for %3d threads\n" $PTHREADS
+    echo -e "==========================\e[39m"
+    for bench in "${uobjects[@]}"; do
+        printf "Validating %-20s \t\t\t\t\t" $bench
+        runs=$(($RUNS_PER_THREAD * $PTHREADS))
```

Comment:
Do you need to recalculate this in the for loop?
Check the same for the other loops.

Reply:
Fixed.